dcm2niix_gear/dcm2niix/arrange.py

```diff
@@ -118,6 +118,121 @@ def exit_if_archive_empty(archive_obj):
         os.sys.exit(1)
 
 
+def tally_files(dir_loc, print_out=False):
+    """
+    Function to profile directory. Note current exclusion of {'.DS_Store', '._*'}
+    from file_set and file_tree. # TODO move this exclusion?
+
+    Args:
+        dir_loc: directory to profile
+        [print_out]: print output
+    Returns:
+        file_set: set of file leaves
+        file_tree
+        file_name_path_dict
+    """
+
+    file_set = set()
+    file_name_path_dict = {}
+    file_tree = []
+    gen = os.walk(dir_loc, topdown=True)
+    next_gen = next(gen)
+    while next_gen != StopIteration:
+
+        # info on current folder and contents
+        (loc_i, dirs_i, files_i) = next_gen
+
+        # populate outputs
+        if len(files_i) > 0:
+
+            # set of files (short paths or leaves) found in directory
+            # TODO could probably just use keys() of dictionary formed below
+            file_set = file_set | set(
+                [file_ij for file_ij in files_i if (
+                    (file_ij != '.DS_Store') &
+                    (file_ij[0:2] != '._'))])
+
+            # directory structure in list form, no filter on files
+            file_tree.append([loc_i + "/" + file_ij for file_ij in files_i])
+
+            for file_ij in files_i:
```

Comment:
I think you could collapse this loop into the one above if you go with a loop approach rather than a list comprehension.

Reply:
Good point, a lot of redundancy there. Rearranged now. Note: in the process moved file_name_path_dict inside the then part of the if regex search statement. Not sure it wasn't covered elsewhere but I think that mighta been leaving the .DS_Store's and ._'s in in some places (my bad).